File: struct2_7_lession_exam/WebContent/login_service.php
<?php session_start()?>
    <!DOCTYPE html>
    <html lang="en">

    <head>
        <meta charset="UTF-8">
        <title>Login Service</title>
    </head>

    <body>
        <?
$code=$_POST["code"];
$pass=$_POST["pass"];
if(!$code||!$pass){
    echo "<script>window.location.href='index.html'</script>";
}else{
    $conn=mysql_connect("localhost","root","root");   
    mysql_selectdb("exam");
    
        $sql="SELECT NAME FROM USER WHERE CODE='".$code."' and pass='".$pass."'";
        $result=mysql_query($sql,$conn);
        $numrows=mysql_num_rows($result);
        if($numrows>0){
            $_SESSION['code']=$code;
            if($row = mysql_fetch_array($result)){
               $_SESSION['name']=$row['NAME'];  
            }
            $sql="SELECT CODE FROM GRADE WHERE CODE='$code'";
            $result=mysql_query($sql,$conn);
            if(mysql_num_rows($result)>0){
                echo "<script>window.location.href='isexam.php'</script>";
                
            }else{
                echo "<script>window.location.href='paper.php'</script>";
            }
        }else{
            echo "<script>window.location.href='index.html'</script>";
        }
}

    
?>
    </body>

    </html>

File: struct2_7_lession_exam/src/cn/com/exam/model/LoginService.java
package cn.com.exam.model;


import java.util.List;
import java.util.Map;

import cn.com.exam.util.DbUtil;

public class LoginService {
	private DbUtil db;
	private List<Map<String,Object>> data;
	private String sql;
	public String login(String code,String pass){
		String result=null;
		try{
			db=new DbUtil();
			sql="select code from user where code=? and pass=?";
			data=db.query(sql, new String[]{code,pass});
			if(data.size()>0){
				result=(String) data.get(0).getOrDefault("name", "");
			}
		}catch( Exception e){
		}
		return result;
	}
}
